test/lib/buildFactory.test.js

```diff
@@ -143,6 +142,17 @@ describe('Build Factory', () => {
             });
         });
 
+        it('ignores extraneous parameters', () => {
+            const expected = {};
+            const garbage = 'garbageData';
+
+            datastore.save.yieldsAsync(null, expected);
+
+            return factory.create({ garbage, username, jobId, sha }).then(() => {
+                assert.calledWith(datastore.save, saveConfig);
```

Comment:
That case was omitted. The datastore is sensitive towards what data it gets, so the intent is to only validate that the `datastore.save` is called with the correct parameters.